pdns/dnsdistdist/dnsdist-configuration-yaml.cc

```diff
@@ -0,0 +1,1593 @@
+          addServerToPool(poolName, downstream);
+        }
+      }
+      else {
+        addServerToPool("", downstream);
+      }
+
+      dnsdist::backend::registerNewBackend(downstream);
+    }
+
+    if (!globalConfig.proxy_protocol.acl.empty()) {
+      dnsdist::configuration::updateRuntimeConfiguration([&globalConfig](dnsdist::configuration::RuntimeConfiguration& config) {
+        config.d_proxyProtocolACL.clear();
+        for (const auto& aclEntry : globalConfig.proxy_protocol.acl) {
+          config.d_proxyProtocolACL.addMask(std::string(aclEntry));
+        }
+      });
+    }
+
+#ifndef DISABLE_CARBON
+    if (!globalConfig.metrics.carbon.empty()) {
+      dnsdist::configuration::updateRuntimeConfiguration([&globalConfig](dnsdist::configuration::RuntimeConfiguration& config) {
+        for (const auto& carbonConfig : globalConfig.metrics.carbon) {
+          auto newEndpoint = dnsdist::Carbon::newEndpoint(std::string(carbonConfig.address),
+                                                          std::string(carbonConfig.name),
+                                                          carbonConfig.interval,
+                                                          carbonConfig.name_space.empty() ? "dnsdist" : std::string(carbonConfig.name_space),
+                                                          carbonConfig.instance.empty() ? "main" : std::string(carbonConfig.instance));
+          config.d_carbonEndpoints.push_back(std::move(newEndpoint));
+        }
+      });
+    }
+#endif /* DISABLE_CARBON */
+
+    if (!globalConfig.webserver.listen_address.empty()) {
+      const auto& webConfig = globalConfig.webserver;
+      loadWebServer(webConfig);
+    }
+
+    if (globalConfig.query_count.enabled) {
+      dnsdist::configuration::updateRuntimeConfiguration([&globalConfig](dnsdist::configuration::RuntimeConfiguration& config) {
+        config.d_queryCountConfig.d_enabled = true;
+        getLuaFunctionFromConfiguration(config.d_queryCountConfig.d_filter, globalConfig.query_count.filter_function_name, globalConfig.query_count.filter_function_code, globalConfig.query_count.filter_function_file, "query count filter function");
+      });
+    }
+
+    loadDynamicBlockConfiguration(globalConfig.dynamic_rules_settings, globalConfig.dynamic_rules);
+
+    if (!globalConfig.tuning.tcp.fast_open_key.empty()) {
+      std::vector<uint32_t> key(4);
+      auto ret = sscanf(globalConfig.tuning.tcp.fast_open_key.c_str(), "%" SCNx32 "-%" SCNx32 "-%" SCNx32 "-%" SCNx32, &key.at(0), &key.at(1), &key.at(2), &key.at(3));
+      if (ret < 0 || static_cast<size_t>(ret) != key.size()) {
+        throw std::runtime_error("Invalid value passed to tuning.tcp.fast_open_key!\n");
+      }
+      dnsdist::configuration::updateImmutableConfiguration([&key](dnsdist::configuration::ImmutableConfiguration& config) {
+        config.d_tcpFastOpenKey = std::move(key);
+      });
+    }
+
+    if (!globalConfig.general.capabilities_to_retain.empty()) {
```

Comment:
I already mentioned it elseweher, most `general` settingss do not seem to be implemented. I only found `capabilities_to_retain` here.

Reply:
Have you seen `dnsdist-rust-lib/dnsdist-configuration-yaml-items-generated.cc`?

Reply:
I'm sure I missed a few settings, to be clear, but I sure hope we are not missing most of them :)

Reply:
Oh, I think I only grepped in the cwd... sorry